platform/ethereum/trustray/transaction.go

```diff
@@ -59,26 +60,35 @@ func AppendTxs(in []blockatlas.Tx, srcTx *Doc, coinIndex uint) (out []blockatlas
 			Value: srcTx.Value,
 		}
 		out = append(out, contractTx)
+		return
 	}
 
 	if len(srcTx.Ops) == 0 {
 		return
 	}
 	op := &srcTx.Ops[0]
-
+	// Token transfer transaction
 	if op.Type == blockatlas.TxTokenTransfer && op.Contract != nil {
-		tokenTx := baseTx
+		var tokenID string
+		switch coinIndex {
+		case coin.WAN:
+			tokenID = address.EIP55ChecksumWanchain(op.Contract.Address)
```

Comment:
Add a helper `func normalizeAddress(string, coin)`? so we don't need to have switch cases in L73 and L122

Reply:
https://github.com/trustwallet/blockatlas/pull/1097/commits/08118e4648ee32e98cb6ee1bc71885b6be1445d9